components/layout_2020/table/mod.rs

```diff
@@ -2,12 +2,66 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
 
-//! HTML Tables (╯°□°)╯︵ ┻━┻.
+//! # HTML Tables (╯°□°)╯︵ ┻━┻
 //!
-//! See <https://html.spec.whatwg.org/multipage/table-processing-model> and
-//! <https://drafts.csswg.org/css-tables>.  This is heavily based on the latter specification, but
-//! note that it is still an Editor's Draft, so there is no guarantee that what is implemented here
-//! matches other browsers or the current specification.
+//! This implementation is based on the [table section of the HTML 5
+//! Specification](https://html.spec.whatwg.org/multipage/#tables), the draft [CSS Table Module Level
+//! 3](https://drafts.csswg.org/css-tables) and the [LayoutNG implementation of
+//! tables](https://source.chromium.org/chromium/chromium/src/third_party/+/main:blink/renderer/core/layout/table)
+//! in Blink. In general, the draft specification differs greatly from what other browsers do, so we
+//! generally follow LayoutNG when in question.
+//!
+//! Table layout is divided into two phases:
+//!
+//! 1. Box Tree Construction
+//! 2. Fragment Tree Construction
+//!
+//! ## Box Tree Construction
+//!
+//! During box tree construction, table layout (`construct.rs`) will traverse the DOM and construct
+//! the basic box tree representation of a table, using the structs defined in this file ([`Table`],
+//! [`TableTrackGroup`], [`TableTrack`], etc). When processing the DOM, elements are handled
+//! differently depending on their `display` value. For instance, an element with `display:
+//! table-cell` is treated as a table cell. These display values are automatically set for HTML
+//! elements by the user agent stylesheet.
```

Comment:
This can be understood as all HTML elements receiving some display in the UA sheet, but e.g. `<span>` just gets the initial `display: inline` without needing the UA stylesheet.

Maybe something like:

> Tabular HTML elements like `<table>` or `<td>` get assigned the corresponding display value in the user agent stylesheet.